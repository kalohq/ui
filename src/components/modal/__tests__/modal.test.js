/* @flow */
/* eslint-env jest */
import React from 'react';
import {shallow} from 'enzyme';
import serializer from 'enzyme-to-json/serializer';

import Modal from '../modal';

expect.addSnapshotSerializer(serializer);

describe('components/modal', () => {
  describe('Modal', () => {
    const create = (spread = {}) => {
      const props = {
        open: true,
        ...spread,
      };
      const element = <Modal {...props} />;
      return {props, element};
    };

    it('should shallow render as expected', () => {
      const {element} = create({
        title: 'Remove freelancer',
      });
      const result = shallow(element);
      expect(result).toMatchSnapshot();
    });

    it('should render with a title', () => {
      const TITLE = 'Remove freelancer';

      const {element} = create({
        title: TITLE,
      });
      const result = shallow(element);
      expect(result.find('H2').contains(TITLE)).toBe(true);
    });
  });
});
