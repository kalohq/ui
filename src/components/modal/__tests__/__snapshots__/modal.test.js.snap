// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/modal Modal should shallow render as expected 1`] = `
<Portal
  static={false}
>
  <HotKeys
    handlers={
      Object {
        "esc": undefined,
      }
    }
  >
    <Styled(PrimitiveBox)
      style={
        Object {
          "zIndex": "",
        }
      }
    >
      <ReactCSSTransitionGroup
        transitionAppear={true}
        transitionAppearTimeout={300}
        transitionEnter={true}
        transitionEnterTimeout={300}
        transitionLeave={true}
        transitionLeaveTimeout={300}
        transitionName="t-slide"
      >
        <Styled(PrimitiveBox)
          aria-labelledby="modalTitle"
          onClick={[Function]}
          role="dialog"
          zIndex=""
        >
          <Styled(PrimitiveBox)
            alignItems="center"
            flexDirection="row"
            justifyContent="center"
            title="Remove freelancer"
          >
            <H2
              id="modalTitle"
            >
              Remove freelancer
            </H2>
          </Styled(PrimitiveBox)>
        </Styled(PrimitiveBox)>
      </ReactCSSTransitionGroup>
    </Styled(PrimitiveBox)>
  </HotKeys>
</Portal>
`;
